### PR TITLE
[INLONG-10328][Manager] Support automatic synchronization of stream fields to sink

### LINKED ISSUE #10328: [Improve][Manager] Support automatic synchronization of stream fields to sink
### Description

Support automatic synchronization of stream fields to sink.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
Fixes #10328 

### Motivation

 Support automatic synchronization of stream fields to sink.

### Modifications

 Support automatic synchronization of stream fields to sink.

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java [hunk 1]
```diff
@@ -59,7 +64,10 @@ public class FieldTypeMappingReader implements Serializable {
     @Getter
     protected final String streamType;
     @Getter
-    protected final Map<String, String> FIELD_TYPE_MAPPING_MAP = Maps.newHashMap();
+    protected final Map<String, String> STREAM_TO_SINK_FIELD_TYPE_MAPPING_MAP = Maps.newHashMap();
+
+    @Getter
+    protected final Map<String, String> SOURCE_TO_SINK_FIELD_TYPE_MAPPING_MAP = Maps.newHashMap();
```
Comment (aloyszhang) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java:70]: why are these two variables named in the upper case?

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/ClickHouseFieldTypeStrategy.java [hunk 2]
```diff
@@ -53,6 +74,6 @@ public String getFieldTypeMapping(String sourceType) {
             }
         }
         String dataType = StringUtils.substringBefore(sourceType, LEFT_BRACKET).toUpperCase();
-        return reader.getFIELD_TYPE_MAPPING_MAP().getOrDefault(dataType, sourceType.toUpperCase());
+        return reader.getSOURCE_TO_SINK_FIELD_TYPE_MAPPING_MAP().getOrDefault(dataType, sourceType.toUpperCase());
```
Comment (aloyszhang) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/ClickHouseFieldTypeStrategy.java:77]: reader.getSOURCE_TO_SINK_FIELD_TYPE_MAPPING_MAP()

Comment (aloyszhang) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/ClickHouseFieldTypeStrategy.java:77]: duplicated with DefaultFieldTypeStrategy#getStreamToSinkFieldTypeMapping

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/DefaultFieldTypeStrategy.java [hunk 3]
```diff
@@ -17,13 +17,50 @@
...
+        }
+        if (StringUtils.isNotBlank(sourceType)) {
+            Matcher matcher = PATTERN.matcher(sourceType.toUpperCase());
+            if (matcher.matches()) {
+                // obtain the field type modified by Nullable, for example, uint8(12) in Nullable(uint8(12))
+                sourceType = matcher.group(1);
+            }
+        }
+        String dataType = StringUtils.substringBefore(sourceType, LEFT_BRACKET).toUpperCase();
+        return reader.getSOURCE_TO_SINK_FIELD_TYPE_MAPPING_MAP().getOrDefault(dataType, sourceType.toUpperCase());
```
Comment (aloyszhang) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/DefaultFieldTypeStrategy.java:64]: getSOURCE_TO_SINK_FIELD_TYPE_MAPPING_MAP -> getSTERAM_TO_SINK_FIELD_TYPE_MAPPING_MAP

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java [hunk 4]
```diff
@@ -20,54 +20,34 @@
...
-        EXTRACT_NODE_PROVIDER_LIST.add(new RedisProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new TubeMqProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new SQLServerProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new PostgreSQLProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new MySQLBinlogProvider());
-        EXTRACT_NODE_PROVIDER_LIST.add(new IcebergProvider());
-
-    }
+    @Autowired
+    private List<ExtractNodeProvider> EXTRACT_NODE_PROVIDER_LIST = new ArrayList<>();
```
Comment (aloyszhang) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java:42]: why use upper case for EXTRACT_NODE_PROVIDER_LIST?

Comment (fuweng11) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/ExtractNodeProviderFactory.java]: Fixed.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java [hunk 5]
```diff
@@ -20,67 +20,34 @@
...
-        LOAD_NODE_PROVIDER_LIST.add(new MySQLProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new OracleProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new PostgreSQLProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new RedisProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new SQLServerProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new StarRocksProvider());
-        LOAD_NODE_PROVIDER_LIST.add(new TDSQLPostgreSQLProvider());
-    }
+    @Autowired
+    private List<LoadNodeProvider> LOAD_NODE_PROVIDER_LIST = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/LoadNodeProviderFactory.java]: Fixed.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/DefaultFieldTypeStrategy.java (hunk 3 above)
Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/DefaultFieldTypeStrategy.java]: Fixed.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/ClickHouseFieldTypeStrategy.java (hunk 2 above)
Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/strategy/ClickHouseFieldTypeStrategy.java]: Fixed.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/fieldtype/FieldTypeMappingReader.java]: Fixed.

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**